packages/core/src/internal/proc.js

```diff
@@ -50,7 +50,9 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
     attaches cancellation logic to this task's continuation
     this will permit cancellation to propagate down the call chain
   **/
-  cont.cancel = task.cancel
+  if (cont !== noop) {
```

Comment:
I would personally prefer:
- removing passing `noop` as an argument to `proc` altogether (making it undefined in cases where `noop` is being passed)
- changing this to just `if (!cont)`
- adding `noop` as default parameter of `cont` in `newTask` [here](https://github.com/redux-saga/redux-saga/blob/374fde6e70945e0d20b2e4c221b74f643e26bd2b/packages/core/src/internal/newTask.js#L9)